### PR TITLE
[ggj][infra][5/5] support goldens updates for AST integration test

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
Find more details in design doc: go/java-micro-file-diff-infra

This PR is for moving the AST integration test to file-diff infrastructure, based one #279

### CODE REVIEW

File: src/test/java/com/google/api/generator/engine/BUILD.bazel [hunk 1]
```diff
@@ -25,3 +27,17 @@ TESTS = [
         "@junit_junit//jar",
     ],
 ) for test_name in TESTS]
+
+TEST_CLASS_NAME = "com.google.api.generator.engine.JavaCodeGeneratorTest"
+
+# Run `bazel run src/test/java/com/google/api/generator/engine:JavaCodeGeneratorTest_update`
+# to update goldens as expected generated code.
+updated_golden(
```
Comment (miraleung) [inline src/test/java/com/google/api/generator/engine/BUILD.bazel:35]: Not part of this PR, but just noticed this. Can we change this (separately) to `update_golden` as was proposed in the original doc?

File: src/test/java/com/google/api/generator/engine/BUILD.bazel [hunk 2]
```diff
@@ -25,3 +27,17 @@ TESTS = [
...
+    name = "JavaCodeGeneratorTest_update",
+    srcs = [
+        ":engine_files",
+        "//src/test/java/com/google/api/generator/engine/goldens:goldens_files",
```
Comment (miraleung) [inline src/test/java/com/google/api/generator/engine/BUILD.bazel:39]: What is the motivation for including all the golden files and all the framework files?

**Review by miraleung — CHANGES_REQUESTED** [review]

File: src/test/java/com/google/api/generator/engine/BUILD.bazel (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply src/test/java/com/google/api/generator/engine/BUILD.bazel:35]: sg, was following Vadym's suggestion to make all the name of bazel target a "noun" instead of "verb". But i'm open to change it back in a separate PR.

File: src/test/java/com/google/api/generator/engine/BUILD.bazel (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply src/test/java/com/google/api/generator/engine/BUILD.bazel:39]: They are needed now, removed. Thanks

File: src/test/java/com/google/api/generator/engine/BUILD.bazel (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply src/test/java/com/google/api/generator/engine/BUILD.bazel:35]: fixed this in PR #357

**Review by miraleung — APPROVED** [review]
